### PR TITLE
Something to do with authorization...its too late for this shit

### PR DESCRIPTION
**Pull request by indiesquidge**

### CODE REVIEW

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class AdminController < ApplicationController
   before_action :authorize_user
 
   def authorize_user
-    if current_user.nil?
+    if current_user.nil? || !current_user.admin?
```
Comment (indiesquidge) [inline app/controllers/admin_controller.rb]: This isn't hacky, amirite?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -29,6 +29,9 @@ def update
   end
 
   def show
+    if current_user.id != params[:id].to_i
```
Comment (indiesquidge) [inline app/controllers/users_controller.rb]: much elegance. Such good code.

File: spec/features/admin_categories_view_spec.rb [hunk 3]
```diff
@@ -3,9 +3,10 @@
 describe "the category view", type: :feature do
   context "as an admin" do
     before :each do
-      @category = Category.create(name: "Food")
+      @category  = Category.create(name: "Food")
       @category2 = Category.create(name: "Water")
-      @admin = FactoryGirl.create(:admin)
+      @admin     = create(:admin)
```
Comment (indiesquidge) [inline spec/features/admin_categories_view_spec.rb]: shut


















UP!

Comment (paulgrever) [inline spec/features/admin_categories_view_spec.rb]: @indiesquidge get your spacing under control bro.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (paulgrever) [inline app/controllers/users_controller.rb]: You have come to the dark side.

**Comment by paulgrever** [comment]
SHIP IT!

**Pull request merged**